src/js/web-components/video-player/VideoPlayer.js

```diff
@@ -111,6 +118,47 @@ export default class extends HTMLElement {
       floatingButtonsBar.appendChild(pipButton);
     }
 
+    this.initializeCast().then((castButton) => {
+      floatingButtonsBar.appendChild(castButton);
+
+      window.cast.framework.CastContext.getInstance().addEventListener(
+        window.cast.framework.CastContextEventType.SESSION_STATE_CHANGED,
+        (e) => {
+          if (e.sessionState === 'SESSION_STARTED' || e.sessionState === 'SESSION_RESUMED') {
+            const castSession = window.cast.framework.CastContext.getInstance().getCurrentSession();
+            const mediaInfo = new window.chrome.cast.media.MediaInfo(
+              this.internal.selectedSource.src,
+              this.internal.selectedSource.type,
+            );
+            const metadata = new window.chrome.cast.media.GenericMediaMetadata();
+
+            metadata.title = videoData.title;
```

Comment:
Shall we fill `images` as well from our Media Session API?
https://developers.google.com/cast/docs/reference/web_receiver/cast.framework.messages.GenericMediaMetadata#title

Reply:
Using the `thumbnail` image instead of the Media Session API images. Changed this in 515e21d.

Or do you think we should rather provide more image sizes since we have them ready anyway? They are all square, though, whereas the thumbnail has landscape aspect ratio.

Reply:
Using media session artworks would allow cast receiver to pick the most appropriate image based on image sizes. We can also leave it as a TODO for improvements. Your call.

Reply:
@dero could you add the TODO in code for future reference

Reply:
@derekherman Added a todo in 1596a09.